core/src/Kokkos_CopyViews.hpp

```diff
@@ -612,12 +613,17 @@ void view_copy(const DstType& dst, const SrcType& src) {
   };
 
   if (!DstExecCanAccessSrc && !SrcExecCanAccessDst) {
-    std::string message(
-        "Error: Kokkos::deep_copy with no available copy mechanism: ");
-    message += src.label();
-    message += " to ";
-    message += dst.label();
-    Kokkos::Impl::throw_runtime_exception(message);
+    std::ostringstream ss;
+    ss << "Error: Kokkos::deep_copy with no available copy mechanism: ";
+    ss << "from src (\"" << src.label() << "\") to dst (\"" << dst.label()
+       << "\").\n";
+    ss << "There is no common execution space that can access both src's "
+          "space\n";
+    ss << "(" << src_memory_space().name() << ") and dst's space ("
+       << dst_memory_space().name() << "), ";
+    ss << "so src and dst\n";
```

Comment:
since we write the message cleanly, I would suggest spelling out "source view" instead of "src" and "destination view" instead of "dst" so that we avoid using names that are specific to the func parameters